ui/pages/token-allowance/token-allowance.js

```diff
@@ -0,0 +1,609 @@
+                fontWeight={FONT_WEIGHT.NORMAL}
+                color={COLORS.TEXT_MUTED}
+              >
+                <span>{t('transactionDetailLayer2GasHeading')}</span>
+                {`${ethTransactionTotal} ${nativeCurrency}`}
+              </Typography>
+            </Box>
+            <MultiLayerFeeMessage
+              transaction={fullTxData}
+              layer2fee={hexTransactionTotal}
+              nativeCurrency={nativeCurrency}
+              plainStyle
+            />
+          </Box>
+        ) : (
+          <>
+            <Box>
+              <Typography
+                variant={TYPOGRAPHY.H7}
+                color={COLORS.TEXT_ALTERNATIVE}
+              >
+                {t('feeAssociatedRequest')}
+              </Typography>
+            </Box>
+            <Box
+              display={DISPLAY.FLEX}
+              flexDirection={FLEX_DIRECTION.COLUMN}
+              alignItems={ALIGN_ITEMS.FLEX_END}
+              textAlign={TEXT_ALIGN.RIGHT}
+            >
+              <Box>
+                <Typography
+                  variant={TYPOGRAPHY.H4}
+                  fontWeight={FONT_WEIGHT.BOLD}
+                  color={COLORS.TEXT_DEFAULT}
+                >
+                  {formatCurrency(fiatTransactionTotal, currentCurrency)}
+                </Typography>
+              </Box>
+              <Box>
+                <Typography
+                  variant={TYPOGRAPHY.H6}
+                  fontWeight={FONT_WEIGHT.NORMAL}
+                  color={COLORS.TEXT_MUTED}
+                >
+                  {`${ethTransactionTotal} ${nativeCurrency}`}
+                </Typography>
+              </Box>
+            </Box>
+          </>
+        )}
+      </Box>
+    );
+  };
+
+  const renderDataContent = () => {
+    return (
+      <Box display={DISPLAY.FLEX} flexDirection={FLEX_DIRECTION.COLUMN}>
+        <Box>
+          <Typography variant={TYPOGRAPHY.H7} color={COLORS.TEXT_ALTERNATIVE}>
+            {isSetApproveForAll
+              ? t('functionSetApprovalForAll')
+              : t('functionApprove')}
+          </Typography>
+        </Box>
+        {isSetApproveForAll && setApproveForAllArg !== undefined ? (
+          <Box>
+            <Typography variant={TYPOGRAPHY.H7} color={COLORS.TEXT_ALTERNATIVE}>
+              {`${t('parameters')}: ${setApproveForAllArg}`}
+            </Typography>
+          </Box>
+        ) : null}
+        <Box
+          marginRight={4}
+          className="token-allowance-container__data__data-block"
+        >
+          <Typography variant={TYPOGRAPHY.H7} color={COLORS.TEXT_ALTERNATIVE}>
+            {data}
+          </Typography>
+        </Box>
+      </Box>
+    );
+  };
+
+  const renderFullDetails = () => {
+    return (
+      <Box
+        display={DISPLAY.FLEX}
+        flexDirection={FLEX_DIRECTION.COLUMN}
+        alignItems={ALIGN_ITEMS.CENTER}
+        className="token-allowance-container__full-tx-content"
```

Comment:
We shouldn't create functions that render more of the tree inside of functional components. Either render it inline in the component or extract it to a new component. When you reach for this it is a good sign that you need to break down your component.

Reply:
Fixed here: db22c9fdcfe133f781d91da0fb9d1b7e0380fccd